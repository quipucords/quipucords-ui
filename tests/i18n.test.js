const { GettextExtractor, JsExtractors } = require('gettext-extractor');

const textExtractor = () => {
  const extractor = new GettextExtractor();
  extractor
    .createJsParser([
      JsExtractors.callExpression(['t', '[this].t'], {
        arguments: {
          text: 0,
          context: 1
        }
      })
    ])
    .parseFilesGlob('./src/components/**/*!(.test|.spec).@(js|jsx)');

  return extractor;
};

describe('i18n locale', () => {
  const getText = textExtractor();

  it('should generate a predictable pot output snapshot', () => {
    expect(getText.getPotString()).toMatchSnapshot('pot output');
  });
});
