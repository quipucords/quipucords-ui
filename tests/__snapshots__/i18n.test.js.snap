// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n locale should generate a predictable pot output snapshot: pot output 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=UTF-8\\\\n\\"

#: src/components/aboutModal/aboutModal.js:99
msgctxt \\"Browser OS\\"
msgid \\"about-modal.browser-os\\"
msgstr \\"\\"

#: src/components/aboutModal/aboutModal.js:93
msgctxt \\"Browser Version\\"
msgid \\"about-modal.browser-version\\"
msgstr \\"\\"

#: src/components/aboutModal/aboutModal.js:105
msgctxt \\"Server Version\\"
msgid \\"about-modal.server-version\\"
msgstr \\"\\"

#: src/components/aboutModal/aboutModal.js:110
msgctxt \\"UI Version\\"
msgid \\"about-modal.ui-version\\"
msgstr \\"\\"

#: src/components/aboutModal/aboutModal.js:89
msgctxt \\"Username\\"
msgid \\"about-modal.username\\"
msgstr \\"\\"
"
`;
